### PR TITLE
Fix marble hover/dragging styles for Firefox

### PR DESCRIPTION
**Pull request by kaplas**
# This PR

- **Replaces CSS filter for marble drop shadows with a native SVG implementation.** #17 

I tried to debug why the previous drop shadow did not work in Firefox. It seems that Firefox handles measurements against SVG viewBox differently than Webkit-based browsers. I did not encounter any exactly same kind of bug report in the Internet, but I did found [this one](https://bugzilla.mozilla.org/show_bug.cgi?id=1150868), where the symptomps were very similar, but the DOM structure did not match. There a commenter actually mentions that Chrome and WebKit would be actually be handling viewBox-based filtering wrongly. Don't know who is right and who is wrong, but at least [by taking into account the number of related FF bugs](https://bugzilla.mozilla.org/buglist.cgi?quicksearch=filter+svg), it is safe to assume that the situation is not going to resolve anytime soon.

The only alternative that I found was to implement a SVG-native drop-shadow filter element, which is then just linked by its ID to the related elements.

- **Hack-fixes `user-select` for Firefox**

For some reason `-moz-user-select` style attribute does not end up to the DOM via `virtual-dom` rendering. That is very weird, taking into account that a similar vendor prefixed version works in Chrome. However it is possible to define that property via a normal CSS file or with the FF CSS dev tool panel. If you have idea what is going on here, please share your wisdom.

As a hack-fix I propose that for this single thing an additional CSS clause could be added to `main.less`.

## I understand 

...that in both of these cases abstraction is leaking from components to the outside world. However, these were the only fixes that I could come up with which reliably fix these FF problems. If you have some ideas on how these fixes should be stored in a more "componentish" way (instead of having those in `index.html` and `main.less`), I'm more than happy to change this PR.

And BTW, after when we have got this PR fixed, I have more global-leakage-related bugs and questions waiting for you; I'm very sorry if I end up tearing apart your component-based dream of the future ;-)

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -11,6 +11,19 @@
...
+        <feGaussianBlur in="SourceAlpha" stdDeviation="0.05"/> <!-- stdDeviation == blur -->
+        <feOffset dx="0" dy="0.05" result="offsetblur"/> <!-- position relative to marble viewBox -->
+        <feFlood flood-color="rgba(0,0,0,0.4)"/>
+        <feComposite in2="offsetblur" operator="in"/>
+        <feMerge>
+          <feMergeNode/>
+          <feMergeNode in="SourceGraphic"/>
+        </feMerge>
+      </filter>
+      </svg>
```
Comment (staltz) [inline index.html]: -2 space indentation for the `</svg>` please

File: index.html [hunk 2]
```diff
@@ -11,6 +11,19 @@
     <script src="dist/js/vendor/modernizr-2.6.2.min.js"></script>
   </head>
   <body>
+    <!-- Cross-browser SVG filter for drop-shadows -->
```
Comment (staltz) [inline index.html]: Have you tried rendering this through the virtual DOM? You could put this near the top-most level inside `.js-appContainer`.

**Comment by staltz** [comment]
Super thanks for the PR and fighting down this bug. 

> For some reason -moz-user-select style attribute does not end up to the DOM via virtual-dom rendering. That is very weird

We need to figure out why. If we can't then we should open an issue on virtual-dom. It "should work".

File: src/components/marble.js [hunk 3]
```diff
@@ -22,18 +22,19 @@ function renderSvg(data, isDraggable, inputStyle, isHighlighted) {
...
       svg('circle', {
-        style: {
-          stroke: Colors.black,
-          fill: color
-        },
+        style: mergeStyles({
+            stroke: Colors.black,
+            fill: color
+          },
+          isDraggable && isHighlighted ? marbleElevation1Style : {}),
```
Comment (staltz) [inline src/components/marble.js]: -2 space indentation for lines 34--37.

**Comment by kaplas** [comment]
@staltz I made a codepen about that `-moz-user-select` + `virtual-dom` problem:
http://codepen.io/cido/pen/rVmyXP?editors=001
The bug seems to be reproducible. As this is literally the first time I'm using `virtual-dom`, could you check that I haven't done any Stupid User Errors there. If it looks legit, I will submit an issue for `virtual-dom`.

**Comment by staltz** [comment]
Seems like there are no Stupid User Errors.

**Comment by kaplas** [comment]
OK, I submitted that one: [virtual-dom/issues/#265](https://github.com/Matt-Esch/virtual-dom/issues/265).

Would you like me to remove that hack-fix, or just add an appropriate todo comment near to the fix, or what?

**Comment by staltz** [comment]
A comment would be enough. :+1:

**Comment by staltz** [comment]
@kaplas I finally had time to work on rxmarbles and merge this. Thanks :+1:

**Comment by kaplas** [comment]
Sorry that I did not have the time to do the changes you requested... :(

**Pull request merged**